zerver/views/users.py

```diff
@@ -203,6 +213,122 @@ def reactivate_user_backend(
 )
 
 
+def filter_changes_in_profile_data(
+    recipient_user: UserProfile,
+    old_profile_data: List[ProfileDataElement],
+    new_profile_data: List[ProfileDataElement],
+) -> Tuple[List[Any], List[Any]]:
+    clean_old_profile_data = []
+    clean_new_profile_data = []
+
+    for field in new_profile_data:
+        id = field["id"]
+        index = id - 1
+        new_value = field["value"] if field["value"] else None
+        old_value = old_profile_data[index]["value"] if old_profile_data[index]["value"] else None
```

Comment:
Since `old_profile_data[index]` is quite long and perhaps difficult to read when used with multiple indexes, maybe set this as a variable named `old_field`